File: shopeasyfrontend/src/services/AdminService.js
import api from "./api";

export const getDashboard = async () => {

    const response = await api.get("/admin/dashboard");

    return response.data;

};

export const getAllUsers = async () => {

    const response = await api.get("/admin/users");

    return response.data;

};

export const getAllOrders = async () => {

    const response = await api.get("/admin/orders");

    return response.data;

};

export const updateOrderStatus = async (id, status) => {

    const response = await api.put(
        `/admin/orders/${id}`,
        {
            status: status
        }
    );

    return response.data;

};
export const enableUser = async (id) => {

    const response = await api.put(`/admin/users/${id}/enable`);

    return response.data;

};

export const disableUser = async (id) => {

    const response = await api.put(`/admin/users/${id}/disable`);

    return response.data;

};

File: shopeasy/src/main/java/com/shopeasy/service/AdminService.java
package com.shopeasy.service;

import java.util.List;

import com.shopeasy.Dto.AdminDashboardDTO;
import com.shopeasy.Dto.OrderDTO;
import com.shopeasy.Dto.UserProfileDTO;
import com.shopeasy.enums.OrderStatus;

public interface AdminService {

    AdminDashboardDTO getDashboard();

    List<OrderDTO> getAllOrders();

    OrderDTO updateOrderStatus(Long orderId, OrderStatus status);

    List<UserProfileDTO> getAllUsers();

    void enableUser(Long userId);

    void disableUser(Long userId);

}